File: leetcode/String/0028_Find_the_Index_of_the_First_Occurrence_in_a_String.cpp
#include <string>
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle.length() == 0)
        	return 0;
        if (haystack.length() < needle.length())
        	return -1;

        int flag = 0;
        for (int i = 0; i <= haystack.length() - needle.length(); i++) {
        	if (haystack.substr(i, needle.length()) == needle)
        		return i;
        }
        return -1;
    }
};

File: leetcode/String/0028_Find_the_Index_of_the_First_Occurrence_in_a_String.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        N, K = len(haystack), len(needle)
        for i in range(N - K + 1):
            if haystack[i:i+K] == needle:
                return i

        return -1
